src/OpenLoco/src/ToolManager.cpp

```diff
@@ -0,0 +1,53 @@
+#include "./ToolManager.h"
+#include "./GameState.h"
+#include <OpenLoco/Interop/Interop.hpp>
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::ToolManager
+{
+    static loco_global<Ui::WindowNumber_t, 0x00523390> _toolWindowNumber;
+    static loco_global<Ui::WindowType, 0x00523392> _toolWindowType;
+    static loco_global<Ui::CursorId, 0x00523393> _toolWindowCursor;
+    static loco_global<int16_t, 0x00523394> _toolWidgetIndex;
+
+    //  0x00523390
+    Ui::WindowNumber_t getToolWindowNumber()
+    {
+        return _toolWindowNumber;
+    }
+    void setToolWindowNumber(Ui::WindowNumber_t toolWindowNumber)
+    {
+        _toolWindowNumber = toolWindowNumber;
+    }
+
+    // 0x00523392
+    Ui::WindowType getToolWindowType()
+    {
+        return _toolWindowType;
+    }
+    void setToolWindowType(Ui::WindowType toolWindowType)
+    {
+        _toolWindowType = toolWindowType;
+    }
+
+    // 0x00523393
+    Ui::CursorId getToolWindowCursor()
+    {
+        return _toolWindowCursor;
+    }
+    void setToolWindowCursor(Ui::CursorId toolWindowCursor)
+    {
+        _toolWindowCursor = toolWindowCursor;
+    }
+
+    // 0x00523394
+    uint16_t getToolWidgetIndex()
```

Comment:
Defined as signed on L12 and unsigned here. -1 or 0xFFFF is used to indicate that the widget index is unset.

Reply:
it was signed in some places I removed the loco_global, and unsigned in others - what should it be? from your comment it sounds like unsigned?

Reply:
Make it unsigned and check all users to make sure they use it expecting that. There is at least one that was doing a < 0 check to look for the 0xFFFF state.

Reply:
good spot, I am actually very unsure about this now. almost every method the widget index is passed into reads the widget index as int16_t. there is one method that even converts it to a uint8_t, and one that checks its bits with a `1 << widgetIndex`. Basically there are 5 or 6 different types its implicitly cast to around the place. i didn't see a < 0 but I did see a >= 0.
Then, in Window.h there is a typedef `using WidgetIndex_t = int8_t` which is different AGAIN! But on that one, perhaps widgetIndex and toolWidgetIndex are truly different types? But I wouldn't have expected that.

Reply:
I've changed it back to a signed `int16_t` as our CI was complaining about a comparison in `MouseInput.cpp` always being true. Could've changed that comparison into checking against 0xFFFF, but making the index signed seems the cleaner solution.